### ISSUE TITLE
C# / Nested Enum / Reflection / EnumDescriptor

### ISSUE THREAD

**Issue by abbotware** [protocolbuffers/protobuf#10696]
I was trying to get the enum descriptors from a nested enum, and it seems it fails due to this old code check:

https://github.com/protocolbuffers/protobuf/blob/488b8b91f81e6f7d50bd5f879ba429d50b0ba901/csharp/src/Google.Protobuf/Reflection/FileDescriptor.cs#L329-L334

(7 years ago @jskeet )

If I step over that check, it works as expected - is there a reason this is even needed?  I would make a PR to remove those few lines, unless there is a good reason to keep them.

**Comment by jskeet** [comment]
That code is *much* more than 7 years old - it goes back to [at least October 2008](https://github.com/jskeet/protobuf-csharp-port/blob/6803686bc06c4d96afd9bd2637f7b37a58596699/src/ProtocolBuffers/Descriptors/FileDescriptor.cs#L158) :)

I'll look into this in more detail later.

**Comment by abbotware** [comment]
@jskeet  for an easy repro of the issue:

https://github.com/abbotware/EnumValueOptions-gRPC-Example/

change: ProfileEnum to a nested enum like

```
message ProfilesEnum {
enum Type  {
	A = 0 [(enum_name) = "AAA"];
	B = 1 [(enum_name) = "BBB"]; 
	C = 2 [(enum_name) = "CCC"];
}
};
```

this line returns null:

`EnumDescriptor ProfilesEnumDescriptor = ProfilesEnumReflection.Descriptor.FindTypeByName<EnumDescriptor>("ProfilesEnum.Type");`

**Comment by jskeet** [comment]
Yup - I need to check what other languages do, and what other impacts this might have.

**Comment by abbotware** [comment]
How is one supposed to get at this metadata at runtime?  I thought that was the whole point of reflection.  If you can demonstrate how to do it, that would be great... otherwise sounds like a bug consistent in 3 implementations :-)

**Comment by abbotware** [comment]
that works for me!  I thought there was no way to get the metadata at all!
